internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -366,6 +369,35 @@ func (r *MachineDeploymentReconciler) MachineSetToDeployments(o client.Object) [
 	return result
 }
 
-func (r *MachineDeploymentReconciler) shouldAdopt(md *clusterv1.MachineDeployment) bool {
+func (r *Reconciler) shouldAdopt(md *clusterv1.MachineDeployment) bool {
 	return !util.HasOwner(md.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
+
+func reconcileExternalTemplateReference(ctx context.Context, c client.Client, apiReader client.Reader, cluster *clusterv1.Cluster, ref *corev1.ObjectReference) error {
```

Comment:
What do you think about moving this func into the `controllers/external` package to avoid duplication?

If I saw correctly it would fit well to funcs like external.Get / external.Delete / external.CloneTemplate

Reply:
currently, we have one of this func for each controller, with slight differences.

TBH I don't mind having MachineSet and MachineDeployment with their own copy, but eventually, we can iterate cleaning up not only the machine deployment controller, but also all the others (in a follow-up PR ).

Reply:
Oh I wasn't aware of the differences. 

Sounds good to me.